File: 1755-defuse-the-bomb/1755-defuse-the-bomb_sliding_window.py
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        result=[0]*len(code)
        if k==0:
            return result
        start,end,windowSum=1,k,0
        if k<0:
            start=len(code)-abs(k)
            end=len(code)-1
        for i in range(start,end+1):
            windowSum+=code[i]
        for i in range(len(code)):
            result[i]=windowSum
            windowSum-=code[start%len(code)]
            windowSum+=code[(end+1)%len(code)]
            start+=1
            end+=1
        return result
        

File: 1755-defuse-the-bomb/1755-defuse-the-bomb-SLIDING-WINDOW.java
class Solution {
    public int[] decrypt(int[] code, int k) {
        int result[]=new int[code.length];
        int start=1,end=k,windowSum=0;
        if(k<0){
            start=code.length-Math.abs(k);
            end=code.length-1;
        }
        for(int i=start;i<end+1;i++){
            windowSum+=code[i];
        }
        for(int i=0;i<code.length;i++){
            result[i]=windowSum;
            windowSum-=code[start%code.length];
            windowSum+=code[(end+1)%code.length];
            start++;
            end++;
        }
        System.out.println(windowSum);
        return result;
    }
}
